src/features/usersSlice.ts

```diff
@@ -0,0 +1,54 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { User } from '../types/User';
+import { client } from '../utils/fetchClient';
+
+export interface UsersState {
+  items: User[];
+  loaded: boolean;
+  hasError: boolean;
+}
+
+const initialState: UsersState = {
+  items: [],
+  loaded: false,
+  hasError: false,
+};
+
+// Asynchronous Thunk for loading users.
+export const loadUsers = createAsyncThunk('users/load', async () => {
+  const response = await client.get<User[]>('/users');
+
+  return response;
+});
+
+const usersSlice = createSlice({
+  name: 'users',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder
+      .addCase(loadUsers.pending, state => {
+        return {
+          ...state,
+          loaded: false,
+          hasError: false,
+        };
+      })
+      .addCase(loadUsers.fulfilled, (state, action) => {
+        return {
+          ...state,
+          items: action.payload,
+          loaded: true,
```

Comment:
The `hasError` property should be reset to `false` in the `fulfilled` case to ensure that any previous errors are cleared when the data is successfully loaded.